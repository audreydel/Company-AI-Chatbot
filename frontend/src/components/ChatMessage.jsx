import PropTypes from "prop-types";

import styles from "./ChatMessage.module.css";

function ChatMessage({ sender, contents, ...props }) {
    if(sender === "user") {
        return (
            <div className={styles.messageContainer} {...props}>
                <div className={styles.senderIconUser} />
                <p className={styles.messageUser}>{ contents }</p>
            </div>
        );
    }
    if(sender === "ai") {
        return (
            <div className={styles.messageContainer} {...props}>
                <div className={styles.senderIconAi} />
                <p className={styles.messageAi}>{ contents }</p>
            </div>
        );
    }

    throw new Error("Invalid message type");
}

ChatMessage.propTypes = {
    sender: PropTypes.string.isRequired,
    contents: PropTypes.string.isRequired
};

export default ChatMessage;
