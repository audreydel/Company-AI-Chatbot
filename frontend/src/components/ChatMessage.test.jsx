import renderer from "react-test-renderer";

import ChatMessage from "./ChatMessage";

describe("Logic tests", () => {
    it("Should throw when the sender is invalid.", () => {
        expect(() => ChatMessage({ sender: "invalid", contents: "contents" })).toThrow(Error);
    });

    it("Should not throw when the sender is valid.", () => {
        expect(() => ChatMessage({ sender: "user", contents: "contents" })).not.toThrow(Error);
        expect(() => ChatMessage({ sender: "ai", contents: "contents" })).not.toThrow(Error);
        expect(() => ChatMessage({ sender: "system", contents: "contents" })).not.toThrow(Error);
    });
});

describe("Snapshot tests", () => {
    it("Should render correctly (user message)", () => {
        const tree = renderer
            .create(<ChatMessage sender="user" contents="user contents" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Should render correctly (ai message)", () => {
        const tree = renderer
            .create(<ChatMessage sender="ai" contents="ai contents" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Should render correctly (system message)", () => {
        const tree = renderer
            .create(<ChatMessage sender="system" contents="system contents" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
