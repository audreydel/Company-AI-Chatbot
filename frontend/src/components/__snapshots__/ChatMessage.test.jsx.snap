// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Should render correctly (ai message) 1`] = `
<div>
  <div
    style={
      {
        "backgroundColor": "#004176",
        "height": "3em",
        "width": "3em",
      }
    }
  >
    <div>
      <img
        src="ai-icon.svg"
      />
    </div>
  </div>
  <p>
    ai contents
  </p>
</div>
`;

exports[`Snapshot tests Should render correctly (system message) 1`] = `
<div>
  <p>
    system contents
  </p>
</div>
`;

exports[`Snapshot tests Should render correctly (user message) 1`] = `
<div>
  <div
    style={
      {
        "backgroundColor": "#012A4E",
        "height": "3em",
        "width": "3em",
      }
    }
  >
    <div>
      <img
        src="user-icon.svg"
      />
    </div>
  </div>
  <p>
    user contents
  </p>
</div>
`;
